### PR TITLE
chore: prepare `dist-typings`

### PR DESCRIPTION
**Pull request by DavideIadeluca**
**Fixes #0000**

**Changes proposed in this pull request:**

**Reviewers should focus on:**

**Screenshot**

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).

**Required changes:**

- [ ] Related [Flarum core extension PR's](https://github.com/flarum/core/pulls): (Omit this section if irrelevant)

### CODE REVIEW

File: js/src/forum/addProfilePane.ts [hunk 1]
```diff
@@ -17,15 +20,15 @@ export default function addProfilePane() {
     },
   };
 
-  extend(UserPage.prototype, 'navItems', function (items) {
-    if (app.forum.attribute('canViewMasquerade') || this.user.canEditMasqueradeProfile()) {
-      const edit = this.user.canEditMasqueradeProfile();
+  extend(UserPage.prototype, 'navItems', function (items: ItemList<Mithril.Children>) {
+    if (app.forum.attribute('canViewMasquerade') || this.user?.canEditMasqueradeProfile()) {
```
Comment (imorland) [inline js/src/forum/addProfilePane.ts:24]:
```suggestion
    if (app.forum.attribute<boolean>('canViewMasquerade') || this.user?.canEditMasqueradeProfile()) {
```
Let's also type the forum attribute

File: js/src/forum/mutateUserHero.tsx [hunk 2]
```diff
@@ -3,18 +3,24 @@ import app from 'flarum/forum/app';
...
+import type ItemList from 'flarum/common/utils/ItemList';
+import type Mithril from 'mithril';
+import type User from 'flarum/common/models/User';
+
 export default function mutateUserHero() {
-  extend(UserCard.prototype, 'infoItems', function (items) {
-    const answers = app.forum.attribute('canViewMasquerade') ? this.attrs.user.bioFields() || [] : [];
+  extend(UserCard.prototype, 'infoItems', function (items: ItemList<Mithril.Children>) {
+    const user = (this.attrs as { user: User }).user;
+    const answers = app.forum.attribute('canViewMasquerade') ? user.bioFields() || [] : [];
```
Comment (imorland) [inline js/src/forum/mutateUserHero.tsx:13]:
```suggestion
    const answers = app.forum.attribute<boolean>('canViewMasquerade') ? user.bioFields() || [] : [];
```

File: js/src/forum/mutateUserHero.tsx [hunk 3]
```diff
@@ -3,18 +3,24 @@ import app from 'flarum/forum/app';
...
 
     items.add(
       'masquerade-bio',
       <div>
         {answers.map((answer) => {
+          console.log('Answer: ', answer);
```
Comment (imorland) [inline js/src/forum/mutateUserHero.tsx:19]:
```suggestion
```
debug code?

File: js/src/forum/panes/ProfilePane.tsx [hunk 4]
```diff
@@ -48,14 +57,17 @@ export default class ProfilePane extends Component {
...
-      this.answers = [];
-      app.store.find('users', this.user.id(), { include: 'masqueradeAnswers' }).then(() => {
-        this.answers = this.user.masqueradeAnswers();
-        m.redraw();
-      });
-    }
+    if (this.answers) return;
+
+    this.answers = [];
+    app.store.find<User>('users', userId, { include: 'masqueradeAnswers' }).then(() => {
```
Comment (imorland) [inline js/src/forum/panes/ProfilePane.tsx:67]: What about converting this entire function to `async`?

**Review by imorland — CHANGES_REQUESTED** [review]
Just a couple of quite minor points, overall is good tho!

File: js/src/forum/panes/ProfilePane.tsx (hunk 4 above)
Comment (DavideIadeluca) [reply js/src/forum/panes/ProfilePane.tsx:67]: Yep, why not. Done in https://github.com/FriendsOfFlarum/masquerade/pull/89/commits/676a9b8fe6ed92f5c4b8ce73d2baa6e2b2579220

**Review by imorland — APPROVED** [review]

**Pull request merged**